File: Sample/ts/src/sco/Int2.ts
import { IScorpioReader } from '../ScorpioProto/Commons/IScorpioReader'
import { ScorpioUtil } from '../ScorpioProto/Commons/ScorpioUtil'
import { TableUtil } from '../ScorpioProto/Table/TableUtil'
import { IData } from '../ScorpioProto/Table/IData'
import { ITable } from '../ScorpioProto/Table/ITable'

export class Int2 implements IData {
    private m_IsInvalid:boolean = false;
    
    private _value1:any = null;
    /*   默认值() */
    getvalue1():number { return this._value1; }
    ID():number { return this._value1; }
    private _value2:any = null;
    /*   默认值() */
    getvalue2():number { return this._value2; }
    
    GetData(key:string):any {
        if ("value1" == key) return this._value1;
        if ("value2" == key) return this._value2;
        return null;
    }
    
    IsInvalid():boolean { return this.m_IsInvalid; }
    private CheckInvalid():boolean {
        return false;
    }
    
    Set(value:Int2) {
        this._value1 = value._value1;
        this._value2 = value._value2;
    }
    
    ToString() {
        return `{
            value1 : ${this._value1}, 
            value2 : ${this._value2}
            }`;
    }
    
    public static Read(fileName:string, reader:IScorpioReader):Int2 {
        var ret = new Int2();
        ret._value1 = reader.ReadInt32();
        ret._value2 = reader.ReadInt32();
        ret.m_IsInvalid = ret.CheckInvalid();
        return ret;
    }
}

File: ScorpioConversion/Sample/go/src/scov/Int2.go
package scov
import (
    "scorpioproto"
)

// Int2 类名
type Int2 struct {
    
    _Value1 int32
    _Value2 int32
}


// ID   默认值()
func (data *Int2) ID() int32 { return data._Value1; }

// GetValue1   默认值()
func (data *Int2) GetValue1() int32 { return data._Value1; }

// GetValue2   默认值()
func (data *Int2) GetValue2() int32 { return data._Value2; }


// GetData 获取数据
func (data *Int2) GetData(key string) interface{} {
    if "Value1" == key {
        return data._Value1;
    }
    if "Value2" == key {
        return data._Value2;
    }
    return nil;
}


// Set 设置数据
func (data *Int2) Set(value *Int2) {
    data._Value1 = value._Value1;
    data._Value2 = value._Value2;
}


// Read 读取数据
func (data *Int2) Read(fileName string, reader scorpioproto.IScorpioReader) {
    data._Value1 = reader.ReadInt32();
    data._Value2 = reader.ReadInt32();
}

// Int2Read 读取数据
func Int2Read(fileName string, reader scorpioproto.IScorpioReader) *Int2 {
    ret := new(Int2)
    ret.Read(fileName, reader)
    return ret
}
